src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.Cancellation.cs

```diff
@@ -100,24 +97,15 @@ public async Task ConnectTimeout_PlaintextStreamFilterTimesOut_Throws(bool useSs
                     handler.ServerCertificateCustomValidationCallback = TestHelper.AllowAllCertificates;
                     var socketsHandler = GetUnderlyingSocketsHttpHandler(handler);
                     socketsHandler.ConnectTimeout = TimeSpan.FromSeconds(1);
+                    socketsHandler.ConnectCallback = (context, token) => new ValueTask<Stream>(new MemoryStream());
```

Comment:
Would this test the connect timeout? In this case, we are already providing "connected" stream, right? e.g. skipping the `ConnectAsync` or `Connect` calls.

Reply:
Yeah, since we are providing a ConnectCallback, we skip doing the actual ConnectAsync.